src/lambda_function/s3_to_glue/app.py

```diff
@@ -12,14 +12,56 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
-def submit_s3_to_json_workflow(messages: list[dict[str,str]], workflow_name):
+
+def filter_object_info(object_info: dict) -> dict:
+    """Filter out object info records that should not be processed. Returns None
+
+       - Records containing owner.txt
+       - Records that don't contain a specific object key like <path_to_file>/<file_name>
+       - Records that are missing a source_key
+       - Records that are missing a bucket_name
+
+    Args:
+        object_info (dict): Object information from source S3 bucket
+
+    Returns:
+        dict: returns back the object info dict if it passes the filter criteria
+            otherwise returns None
+    """
+    if not object_info["source_key"]:
+        logger.info(
+            "This object_info record doesn't contain a source key "
```

Comment:
nit: we may want to think about how these messages look like in cloudtrail.  We should be able to mine these messages easily to monitor this.

Reply:
Agreed, I wonder if we should create an exception class/tag for just custom exceptions logged by us. Something like this could be NoSourceKeyException, and we define that somewhere. Or, we just mark all of our custom logging with SQSToLambdaWarning: <some_message> so we could easily do a search for these.

Reply:
Ya, I recommend having json structures, because there are technologies that can parse those out into tabular format.  I'll create a Jira ticket to track

Reply:
I think it's premature to format log messages as JSON before we have a schema for our log messages. If the idea is to have these log messages be consumed by something like snowflake we ought to understand what information we want to include in every log message.

Reply:
Going off of what you had suggested in the past, if we logged each record processed, we can potentially have a comparison of records in parquet vs records in JSON.  Agreed that determining a schema for log messages is important.